### PR TITLE
subscription setup

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
If you could pay some attention to the technical aspects, too, I'd appreciate it :)

### CODE REVIEW

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
+==title==
+GrapQL subscriptions with Absinthe - the set-up
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]:
```suggestion
GraphQL subscriptions with Absinthe - the setup
```

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+==author==
+Cornelia Kelinske
+
+==description==
+This is part 1 of a 3-part series on setting up and testing GraphQL subscriptions with Absinthe.
+
+==tags==
+coding, elixir, graphql, absinthe, subscriptions
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md:11]: I don't like the coding tag... it is very "code monkey" feeling.

Is there an alternative tag?

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+==body==
+
+# 1. Why am I writing this post?
+
+I have been taking the (Learn Elixir)[https://learn-elixir.dev/] course, where a lot of the assignments involve GraphQL and Absinthe. And what can I say? I like it. However, the one part that I found a bit more difficult was dealing with subscriptions.
+While writing and testing queries and mutations is well-documented and discussed in a number of resources, information on setting up and, in particular, testing subscriptions is harder to come by. I have spent quite a bit of time on this topic and, as a result, have developed a solid love-hate relationship with those subscriptions. 
+But to answer the question as to why am I writing this post: for future reference! Everything I know about subscriptions in one place. So let's get started!
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]: Why are there new lines here. Is it supposed to be the same paragraph? Wondering if deliberate

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+│   │   └── user.ex
+│   ├── queries
+│   │   ├── some_query.ex
+│   │   └── user.ex
+│   └── subscriptions
+│       └── user.ex
+|── types
+|   ├── some_type.ex
+|   └── user.ex 
+├── schema.ex
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md:50]: so nice

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+With this out of the way, let's get started!
+
+
+# 3. Subscription set-up - the infrastructure
+
+As in most cases, it is worth checking out the [hex.docs](https://hexdocs.pm/absinthe/subscriptions.html) first.
+Based on those and on what I have learned in the course, plus some personal experience, I have come up with the following steps for the basic setup:
+
+First, we need to add some dependencies (if we are running an app as described in 2. above, we'll already have those):
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]:
```suggestion
First, we need to add some dependencies to `mix.exs` (if we are running an app as described in 2. above, we'll already have those): 
```

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+```elixir
+defmodule MyAppWeb.UserSocket do
+  use Phoenix.Socket
+  use Absinthe.Phoenix.Socket,
+   schema: MyAppWeb.Schema
+ def connect(_params, socket) do
+  {:ok, socket}
+ end
+ def id(_socket), do: nil
+end
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]: Want to put some spaces in here between things?

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  longpoll: false
+```
+
+At this point, we should also make sure our `UserSocket` is included in our router for our GraphiQL route like so:
+
+```elixir
+if Mix.env() === :dev do
+  forward "/graphiql", Absinthe.Plug.GraphiQL,
+    schema: GraphqlApiWeb.Schema,
+    socket: GraphqlApiWeb.UserSocket,
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]:
```suggestion
    schema: MyAppWeb.Schema,
    socket: MyAppWeb.UserSocket,
```

File: posts/2022/10-11-graphql_subscription_setup.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+  @moduledoc false
+  use Absinthe.Schema  
+
+  import_types MyAppWeb.Types.SomeType
+  import_types MyAppWeb.Types.User 
+  import_types GraphqlApiWeb.Schema.Queries.SomeQuery  
+  import_types GraphqlApiWeb.Schema.Queries.User  
+  import_types GraphqlApiWeb.Schema.Mutations.SomeMutation
+  import_types GraphqlApiWeb.Schema.Mutations.User 
+  import_types GraphqlApiWeb.Schema.Subscriptions.User
```
Comment (alanvardy) [inline posts/2022/10-11-graphql_subscription_setup.md]:
```suggestion
  import_types MyAppWeb.Schema.Queries.SomeQuery  
  import_types MyAppWeb.Schema.Queries.User  
  import_types MyAppWeb.Schema.Mutations.SomeMutation
  import_types MyAppWeb.Schema.Mutations.User 
  import_types MyAppWeb.Schema.Subscriptions.User
```

**Review by alanvardy — APPROVED** [review]
This is really useful.

Also, do you know that you called the PR "random test" ?

File: posts/2022/10-11-graphql_subscription_setup.md (hunk 1 above)
Comment (CorneliaKelinske) [reply posts/2022/10-11-graphql_subscription_setup.md]: > This is really useful.
> 
> Also, do you know that you called the PR "random test" ?

That is very embarrassing and it is called so no longer

File: posts/2022/10-11-graphql_subscription_setup.md (hunk 4 above)
Comment (CorneliaKelinske) [reply posts/2022/10-11-graphql_subscription_setup.md:50]: Thank you! :)

File: posts/2022/10-11-graphql_subscription_setup.md (hunk 6 above)
Comment (CorneliaKelinske) [reply posts/2022/10-11-graphql_subscription_setup.md]: I suppose :)

**Pull request merged**